File: model/dictionary/ProductRest.js
/**
 * Классы и типы для работы с информацией о товарных остатках.
 * @module module:productRest
 */
import ProductInfo from "./ProductInfo"

/**
 * @class module:productRest.ProductRest
 * @classdesc Остатки продукции на складе и магазине с количествами.
 * @param {module:productInfo.ProductInfo} productInfo Информация о продукции
 * @param {number} stockQuantity Количество продукции на складе
 * @param {number} shopQuantity Количество продукции в торговом зале
 */
export default class ProductRest {
    constructor(productInfo: ProductInfo,
                stockQuantity: number,
                shopQuantity: number) {
        this.productInfo = productInfo;
        this.stockQuantity = stockQuantity;
        this.shopQuantity = shopQuantity;
    }
}

/**
 * @class module:productRest.ShopCommodity
 * @classdesc Позиция на остатках в магазине.
 * @param {number} quantity Количество
 * @param {module:productInfo.ProductInfo} productInfo Информация о продукции
 */
export class ShopCommodity {
    constructor(quantity: number,
                productInfo: ProductInfo) {
        this.quantity = quantity;
        this.productInfo = productInfo;
    }
}

/**
 *
 * @class module:productRest.StockCommodity
 * @classdesc Позиция на остатках на складе.
 * @param {string} informF1RegId Регистрационный номер раздела справки 1
 * @param {string} informF2RegId Регистрационный номер раздела справки 2
 * @param {number} quantity Количество
 * @param {module:productInfo.ProductInfo} productInfo Информация о продукции
 */
export class StockCommodity {
    constructor(informF1RegId: string,
                informF2RegId: string,
                quantity: number,
                productInfo: ProductInfo) {
        this.informF1RegId = informF1RegId;
        this.informF2RegId = informF2RegId;
        this.quantity = quantity;
        this.productInfo = productInfo;
    }
}

File: android/src/main/java/com/evotor/egais/mapper/ProductRest.kt
package com.evotor.egais.mapper

import android.database.Cursor
import com.evotor.egais.converter.QuantityBigDecimalConverter
import com.facebook.react.bridge.Arguments
import com.facebook.react.bridge.WritableMap

internal fun createProductRest(cursor: Cursor): WritableMap {
    val result = Arguments.createMap()
    result.putMap("productInfo", createProductInfo(cursor))
    result.putDouble("stockQuantity", QuantityBigDecimalConverter.toBigDecimal(cursor.getLong(cursor.getColumnIndexOrThrow("STOCK_QUANTITY"))).toDouble())
    result.putDouble("shopQuantity", QuantityBigDecimalConverter.toBigDecimal(cursor.getLong(cursor.getColumnIndexOrThrow("SHOP_QUANTITY"))).toDouble())
    result.putDouble("totalQuantity", QuantityBigDecimalConverter.toBigDecimal(cursor.getLong(cursor.getColumnIndexOrThrow("TOTAL_QUANTITY"))).toDouble())
    return result
}

internal fun createShopCommodity(cursor: Cursor): WritableMap {
    val result = Arguments.createMap()
    result.putDouble("quantity", QuantityBigDecimalConverter.toBigDecimal(cursor.getLong(cursor.getColumnIndexOrThrow("QUANTITY"))).toDouble())
    result.putMap("productInfo", createProductInfo(cursor))
    return result
}

internal fun createStockCommodity(cursor: Cursor): WritableMap {
    val result = Arguments.createMap()
    result.putString("informF1RegId", cursor.getString(cursor.getColumnIndexOrThrow("INFORM_F1_REG_ID")))
    result.putString("informF2RegId", cursor.getString(cursor.getColumnIndexOrThrow("INFORM_F2_REG_ID")))
    result.putDouble("quantity", QuantityBigDecimalConverter.toBigDecimal(cursor.getLong(cursor.getColumnIndexOrThrow("QUANTITY"))).toDouble())
    result.putMap("productInfo", createProductInfo(cursor))
    return result
}